apis/python/src/tiledbsoma/options/_soma_tiledb_context.py

```diff
@@ -146,6 +151,15 @@ def timestamp(self) -> Optional[datetime.datetime]:
             return None
         return ms_to_datetime(self.timestamp_ms)
 
+    @property
+    def native_context(self) -> clib.SOMAContext:
+        """The C++ SOMAContext for this SOMA context."""
+        if self._native_context is None:
```

Comment:
This is lock-free, but lazily instantiated.  does it need a concurrency guard like that used in `tiledb_ctx()` below?